File: ejercicios python 2/ejercicio 6.py
def validacion_token(token):
    if len(token) >= 12:
        return True
    elif len(token) <= 12:
        return False
    
    tiene_numero = False
    
    for caracter in token:
        if caracter.isdigit():
            tiene_numero = True
            break
    
    if not tiene_numero:
        return False
    
    if token.startwith("TEST"):
        return False

    return True


# -------------------------
# PROGRAMA PRINCIPAL
# -------------------------

token_usuario = input("Ingresa el token: ")

resultado = validacion_token(token_usuario)

print("¿Token válido?:", resultado)      

File: ejercicios java semana 2/ejercicio 6.js
const readline = require('readline');

// Creamos la interfaz para leer la entrada del usuario por consola
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

/**
 * Función de validación de token
 */
function validacionToken(token) {
    // Nota: He mantenido tu lógica original, pero revisa la sección de abajo
    if (token.length >= 12) {
        return true;
    } else if (token.length <= 12) {
        return false;
    }

    // --- CÓDIGO INALCANZABLE EN TU VERSIÓN ORIGINAL ---
    let tieneNumero = false;

    for (let caracter of token) {
        if (/\d/.test(caracter)) { // Comprueba si es un dígito
            tieneNumero = true;
            break;
        }
    }

    if (!tieneNumero) {
        return false;
    }

    if (token.startsWith("TEST")) {
        return false;
    }

    return true;
}

// -------------------------
// PROGRAMA PRINCIPAL
// -------------------------
rl.question('Ingresa el token: ', (tokenUsuario) => {
    const resultado = validacionToken(tokenUsuario);
    console.log("¿Token válido?:", resultado);
    rl.close();
});